applications/dds_video/dds_video.cpp

```diff
@@ -0,0 +1,259 @@
+/**
+ * @brief Application to publish a V4L2 video stream to DDS.
+ */
+class V4L2ToDDS : public holoscan::Application {
+ public:
+  explicit V4L2ToDDS(uint32_t domain_id, uint32_t stream_id)
+      : domain_id_(domain_id), stream_id_(stream_id) {}
+
+  void compose() override {
+    using namespace holoscan;
+
+    auto v4l2 = make_operator<ops::V4L2VideoCaptureOp>("v4l2",
+        Arg("allocator") = make_resource<UnboundedAllocator>("pool"),
+        Arg("width", 640u), Arg("height", 480u));
+
+    auto dds = make_operator<ops::DDSVideoPublisherOp>("dds",
+        Arg("participant_qos", std::string("HoloscanDDSTransport::SHMEM+LAN")),
+        Arg("writer_qos", std::string("HoloscanDDSDataFlow::Video")),
+        Arg("domain_id", domain_id_),
+        Arg("stream_id", stream_id_));
+
+    add_flow(v4l2, dds, {{"signal", "input"}});
+  }
+
+ private:
+  uint32_t domain_id_;
+  uint32_t stream_id_;
+};
+
+/**
+ * @brief Application to render a DDS video stream (published by the DDSVideoPublisher)
+ * and shapes (published by the RTI Connext Shapes Demo) to Holoviz.
+ */
+class DDSToHoloviz : public holoscan::Application {
+ public:
+  explicit DDSToHoloviz(uint32_t domain_id, uint32_t stream_id)
+      : domain_id_(domain_id), stream_id_(stream_id) {}
+
+  void compose() override {
+    using namespace holoscan;
+
+    std::shared_ptr<UnboundedAllocator> allocator = make_resource<UnboundedAllocator>("pool");
+
+    //  DDS Video Subscriber
+    auto participant_qos = std::string("HoloscanDDSTransport::SHMEM+LAN");
+    auto video_subscriber = make_operator<ops::DDSVideoSubscriberOp>("video_subscriber",
+        Arg("allocator", allocator),
+        Arg("domain_id", domain_id_),
+        Arg("stream_id", stream_id_),
+        Arg("participant_qos", participant_qos),
+        Arg("reader_qos", std::string("HoloscanDDSDataFlow::Video")));
+
+    // DDS Shapes Subscriber
+    auto shapes_subscriber = make_operator<ops::DDSShapesSubscriberOp>("shapes_subscriber",
+        Arg("domain_id", domain_id_),
+        Arg("participant_qos", participant_qos),
+        Arg("reader_qos", std::string("HoloscanDDSDataFlow::Shapes")));
+
+    // DDS Shapes Renderer
+    auto shapes_renderer = make_operator<ops::DDSShapesRenderer>("shapes_renderer",
+        Arg("allocator", allocator));
+
+    // Holoviz (initialize with the default input spec for the video stream)
+    std::vector<ops::HolovizOp::InputSpec> input_spec;
+    auto& video_spec = input_spec.emplace_back(
+        ops::HolovizOp::InputSpec("", ops::HolovizOp::InputType::COLOR));
+    auto holoviz = make_operator<ops::HolovizOp>("holoviz",
+        Arg("width", 640u), Arg("height", 480u), Arg("tensors", input_spec));
+
+    add_flow(video_subscriber, holoviz, {{"output", "receivers"}});
+    add_flow(shapes_subscriber, shapes_renderer, {{"output", "input"}});
+    add_flow(shapes_renderer, holoviz, {{"outputs", "receivers"}});
+    add_flow(shapes_renderer, holoviz, {{"output_specs", "input_specs"}});
```

Comment:
I'm actually not sure, could these be combined into one line or are two `add_flow`s required?

Reply:
Just tried it and yes, it can. The `holoviz_geometry` example in the SDK keeps them separate so I had just copied that. Updated it to use a single call.